<Action id="21821" issue="15517" author="camelbeck" type="comment" created="2017-04-03 01:29:31.0" updateauthor="camelbeck" updated="2017-04-03 01:29:31.0"> <body><! CDATA During the process of testing (run "make unittest"), it is often (int most cases) stuck in test/unit/keyvaluestore_test.py execution. Even before enable code coverage. Thus after I commit the change set, the jobbuilder thought it is an failure when this situation happens. I will try to attach this console log from jobbuilder.  I wonder that whether this situation happens before. Anyway, I think it is a problem, and I will try to find the reason of it and fix it if it is a defect.  Any suggestions for this issue, please comment here.  ></body> </Action>
<Action id="21937" issue="15517" author="baohua" type="comment" created="2017-04-07 03:22:25.0" updateauthor="baohua" updated="2017-04-07 03:22:25.0"> <body><! CDATA Thanks for the detailed explanation.  Let's record the stuck problem, and maybe can raise a separate bug in jira tasks.  ></body> </Action>
<Action id="22925" issue="15517" author="camelbeck" type="comment" created="2017-04-24 03:32:22.0" updateauthor="camelbeck" updated="2017-04-24 03:32:22.0"> <body><! CDATA Issue: Invocation Error after enabling coverage test.  Detailed description:  Someone encounters following error when run command "make unittest" after enabling coverage test. The prompted error message tells that py.test command cannot find --cov flag. !screenshot-1.png|thumbnail!   Cause explained: Firstly there is nothing wrong with our SDK  and our testing mechanism. It is related with the implementation of tox. When executed, tox will first try to find configure file under the virtualenv folder by default. If it already exists, tox will not create the virtualenv any more no matter whether all the dependencies have already existed. And our testing process has used the default setting of tox. As in the work of enabling the coverage test, we have added the plugin pytest-cov. Thus former users who have already run the "make unittest" may encounter such issue.  Solution: Run "make clean" first.  ></body> </Action>
<Action id="22926" issue="15517" author="baohua" type="comment" body="Very clear reason and solution, thanks, chao!" created="2017-04-24 03:36:58.0" updateauthor="baohua" updated="2017-04-24 03:36:58.0"/>
